src/ts/spatialnavigation/spatialnavigation.ts

```diff
@@ -0,0 +1,170 @@
+import { NavigationGroup } from './navigationgroup';
+import { RootNavigationGroup } from './rootnavigationgroup';
+import { NodeEventSubscriber } from './nodeeventsubscriber';
+import { SeekBarHandler } from './seekbarhandler';
+
+export enum Direction {
+  UP = 'up',
+  DOWN = 'down',
+  LEFT = 'left',
+  RIGHT = 'right',
+}
+
+export enum Action {
+  SELECT = 'select',
+  BACK = 'back',
+}
+
+const KeyMap: {[key: number]: Direction | Action} = {
```

Comment:
where does that keymap come from? Which devices can we expect it to work with? (maybe add a comment)

Reply:
Restructured the key maps in c52e166b 👍

Reply:
We have refactored it to have different key groups for different devices - will give a bit more insight.
Addressed in [c52e166](https://github.com/bitmovin/bitmovin-player-ui/pull/468/commits/c52e166b1316399acc965b1f8c306f9e44967912)